modules/trusted-firmware-m/Kconfig.tfm

```
@@ -276,14 +276,54 @@ config TFM_MCUBOOT_DATA_SHARING
 
 endif # TFM_BL2
 
+choice TFM_MODEL
+	prompt "TF-M Firmware Framework model" if TFM_PROFILE_TYPE_NOT_SET
+	default TFM_IPC
+	help
+	  The Firmware Framework M (FF-M) provides different programming models
+	  for Secure Partitions.
+
+config TFM_LIBRARY
+	bool "Library Model"
+	help
+	  When enabled this option signifies that the TF-M build use secure
+	  library model instead of PSA API.
+
+	  The library model conforms to the PSA Firmware Framework for M (FF-M)
+	  v1.0.
+
+	  NOTE: The Library Model will be removed and replaced by the SFN model.
+
 config TFM_IPC
-	bool "IPC" if TFM_PROFILE_TYPE_NOT_SET
-	default y if (TFM_PROFILE_TYPE_MEDIUM || TFM_PROFILE_TYPE_LARGE)
+	bool "IPC Model"
+	help
+	  When enabled, this option signifies that the TF-M build supports
+	  the PSA API using the IPC Model as the SPM backend.
+	  The IPC model support the IPC and SFN Partition models, and the
+	  isolation levels 1, 2 and 3.
+
+	  In this model the Secure Partition processes signals in any order,
+	  and can defer responding to a message while continuing to process
+	  other signals.
+
+	  The IPC model conforms to the PSA Firmware Framework for M (FF-M)
+	  v1.1.
+
+config TFM_SFN
+	bool "SFN model"
 	help
 	  When enabled, this option signifies that the TF-M build supports
-	  the PSA API (IPC mode) instead of the secure library mode. When
-	  TF-M Profile option is supplied, do not allow manual setting of
-	  the IPC mode, as it is determined by the profile setting.
+	  the PSA API using the SFN Model as the SPM backend.
+	  The SFN model support the SFN Partition model, and the isolation
+	  levels 1.
```

Super nit, but since there is a CI failure we may as well fix this at the same time as any update:

```suggestion
	  The SFN model supports the SFN Partition model, and isolation
	  level 1.
```

---

Spelling is Kconfig help text is not nit ;)